pkg/controllers/cloudmap_controller.go

```diff
@@ -303,49 +308,30 @@ func createEndpointSlicesStruct(svcImport *v1alpha1.ServiceImport, cloudMapSvc *
 }
 
 func extractServicePorts(svc *model.Service) []v1.ServicePort {
-	ports := extractPorts(svc)
-
-	servicePorts := make([]v1.ServicePort, 0, len(ports))
-	for _, p := range ports {
-		servicePorts = append(servicePorts, v1.ServicePort{
-			Protocol: v1.ProtocolTCP,
-			Port:     p,
-		})
+	uniquePorts := make(map[int32]model.Port)
```

Comment:
Are we supporting TCP ports only? UDP and TCP ports are distinct. ie. you can have separate listeners on UDP port 80 and TCP port 80 on a given IP without conflict.

Reply:
We are adding the support for all the ports in this PR. That's a good point, I will add TCP to the uniqueness check.